src/components/navigation.tsx

```diff
@@ -9,27 +9,44 @@ type Props = {
 };
 
 const Navigation = ({ sections, currentSlug }: Props) => {
-  const [isOpen, setIsOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
   const toggle = () => setIsOpen(!isOpen);
   const onItemClick = () => {
     if (isSmallScreen()) {
       toggle();
     }
   };
   const className = isOpen ? 'side-nav side-nav--open' : 'side-nav';
+  let flatSections: NavigationSectionItem[] = [];
+  Object.keys(sections).map((sectionKey: string) => {
+    flatSections = [...flatSections, ...sections[sectionKey]];
+  });
+
+  let currentSlugIndex: number = -1;
+  for (let i: number = 0; i < flatSections.length; i++) {
+    if (flatSections[i].slug === currentSlug) {
+      currentSlugIndex = i;
+      break;
+    }
+  }
```

Comment:
nit: This whole block can be simplified with the `findIndex` method.

```diff
-  let currentSlugIndex: number = -1;
-  for (let i: number = 0; i < flatSections.length; i++) {
-    if (flatSections[i].slug === currentSlug) {
-      currentSlugIndex = i;
-      break;
-    }
-  }
+  const currentSlugIndex: number = flatSections.findIndex(
+    flatSection => flatSection.slug === currentSlug
+  );
```

Reply:
@sagirk, similar to `.find`, I was using `. findIndex` in until [
328da61](https://github.com/nodejs/nodejs.dev/pull/187/commits/328da61741eb03b94e0bb96f6ec105f891e7290b), because I saw lack of IE support for [`.find` Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex).

Reply:
See above.